classes/LocalizationPack.php

```diff
@@ -352,7 +352,20 @@ protected function _installCurrencies($xml, $install_mode = false)
                         return false;
                     }
                     Cache::clear();
+                    // set default data from CLDR
                     $this->setCurrencyCldrData($currency, $defaultLang);
+
+                    // Following is required when installing new currency on existing languages:
+                    // we want to properly update the symbol in each language
+                    $languages = Language::getLanguages();
+                    $context = Context::getContext();
+                    $container = isset($context->controller) ? $context->controller->getContainer() : null;
+                    if (null === $container) {
+                        $container = SymfonyContainer::getInstance();
+                    }
+                    /** @var LocaleRepository $localeRepoCLDR */
+                    $localeRepoCLDR = $container->get('prestashop.core.localization.cldr.locale_repository');
```

Comment:
You already have this code in the file. Can you please refacto in a common method?